dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java

```diff
@@ -174,19 +178,70 @@ public CollectionRest findOne(Context context, UUID id) {
                     CommunityRest.CATEGORY + "." + CommunityRest.NAME + " with id: " + communityUuid
                         + " not found");
             }
-            List<Collection> collections = cs.findAuthorized(context, com, Constants.ADD);
-            return converter.toRestPage(collections, pageable, utils.obtainProjection());
-        } catch (SQLException e) {
+            List<Collection> collections = new LinkedList<Collection>();
+            DiscoverResult resp = discoverAuthorizedCollections(pageable, q, context, com);
+            long tot = resp.getTotalSearchResults();
+            for (IndexableObject solrCollections : resp.getIndexableObjects()) {
+                Collection c = ((IndexableCollection) solrCollections).getIndexedObject();
+                collections.add(c);
+            }
+            return converter.toRestPage(collections, pageable, tot , utils.obtainProjection());
+        } catch (SQLException | SearchServiceException e) {
             throw new RuntimeException(e.getMessage(), e);
         }
     }
 
+    private DiscoverResult discoverAuthorizedCollections(Pageable pageable, String q, Context context, Community com)
```

Comment:
This entire method seems like we may want to consider moving it into the `dspace-api`.  It's a bit odd to me to have a RestRepository class building detailed Solr queries, which makes me wonder if we need to consider moving this into the `CollectionIndexFactoryImpl` or somewhere similar (and then just calling that new method from this RestRepository class)?

Reply:
you are right. Should we move it to the `CollectionService`? should we name it findAuthorizedCollectionsInSOLR

Reply:
@Micheleboychuk please refactor the code adding the method in the CollectionService as by comment above. We have discussed that briefly in yesterday meeting and @tdonohue was just fine to move it in the dspace-api layer